platform/legacy/LegacyPlatformProxifier.ts

```diff
@@ -54,23 +54,39 @@ export class LegacyPlatformProxifier extends EventEmitter {
   /**
    * Neither new nor legacy platform should use this method directly.
    */
-  async listen(port: number, host: string, callback?: () => void) {
+  async listen(port: number, host: string, callback?: (error?: Error) => void) {
     this.log.debug(`"listen" has been called (${host}:${port}).`);
 
-    await this.root.start();
+    let error: Error | undefined;
+    try {
+      await this.root.start();
+    } catch (err) {
+      error = err;
+      this.emit('error', err);
```

Comment:
That's the main thing in this PR - if we properly imitate HttpServer behavior, Hapi will do the rest of the job of crashing Kibana for us, like it does in the legacy platform.